ecs-cli/modules/cli/local/up_app.go

```diff
@@ -14,18 +14,182 @@
+		SecretId: aws.String(arn),
+	})
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to retrieve decrypted secret from %s due to %v", arn, err)
+	}
+	return *val.SecretString, nil
+}
+
 // Up creates a Compose file from an ECS task definition and runs it locally.
 //
 // The Amazon ECS Local Endpoints container needs to be running already for any local ECS task to work
 // (see https://github.com/awslabs/amazon-ecs-local-container-endpoints).
 // If the container is not running, this command creates a new network for all local ECS tasks to join
 // and communicate with the Amazon ECS Local Endpoints container.
 func Up(c *cli.Context) {
-	client := docker.NewClient()
-	network.Setup(client)
+	// TODO When we don't provide any flags, Create() should just check if a "./docker-compose.local.yml" already exists.
+	// If so then Create() should do nothing else, otherwise it should error.
+	Create(c)
+
+	network.Setup(docker.NewClient())
+
+	config := readComposeFile(c)
+	secrets := readSecrets(config)
+	envVars := decryptSecrets(secrets)
+	upComposeFile(config, envVars)
+}
+
+func readComposeFile(c *cli.Context) *composeV3.Config {
+	filename := localproject.LocalOutDefaultFileName
+	if c.String(flags.LocalOutputFlag) != "" {
+		filename = c.String(flags.LocalOutputFlag)
+	}
+	config, err := converter.UnmarshalComposeFile(filename)
+	if err != nil {
+		logrus.Fatalf("Failed to unmarshal Compose file %s due to %v", filename, err)
+	}
+	return config
+}
+
+func readSecrets(config *composeV3.Config) []*secrets.ContainerSecret {
+	var containerSecrets []*secrets.ContainerSecret
+	for _, service := range config.Services {
+		for label, secretARN := range service.Labels {
+			if !strings.HasPrefix(label, converter.SecretLabelPrefix) {
+				continue
+			}
+			namespaces := strings.Split(label, ".")
+			secretName := namespaces[len(namespaces)-1]
+
+			containerSecrets = append(containerSecrets, secrets.NewContainerSecret(service.Name, secretName, secretARN))
+		}
+	}
+	return containerSecrets
+}
+
+func decryptSecrets(containerSecrets []*secrets.ContainerSecret) (envVars map[string]string) {
+	ssmClient, err := newSSMDecrypter()
+	secretsManagerClient, err := newSecretsManagerDecrypter()
+	if err != nil {
+		logrus.Fatal(err.Error())
+	}
+
+	envVars = make(map[string]string)
+	for _, containerSecret := range containerSecrets {
+		service, err := containerSecret.ServiceName()
+		if err != nil {
+			logrus.Fatalf("Failed to retrieve the service of the secret due to %v", err)
+		}
+
+		decrypted := ""
+		err = nil
+		if service == secretsmanager.ServiceName {
+			decrypted, err = containerSecret.Decrypt(secretsManagerClient)
+		}
+		if service == ssm.ServiceName {
+			decrypted, err = containerSecret.Decrypt(ssmClient)
+		}
+		if err != nil {
+			logrus.Fatal(err.Error())
+		}
+		envVars[containerSecret.Name()] = decrypted
+	}
+	return
+}
+
+func upComposeFile(config *composeV3.Config, envVars map[string]string) {
```

Comment:
Even though this is not an exported function I wonder if it's worth noting (via comment, or a more descriptive name?) that this is combining env vars into the converted compose file?